fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/Value.java

```diff
@@ -448,18 +455,27 @@ static List<Value> fromKeyExpressions(@Nonnull final Collection<? extends KeyExp
     }
 
     /**
-     * A scalar value type that cannot be evaluated.
+     * A scalar {@link Value} that cannot be evaluated.
      */
     @API(API.Status.EXPERIMENTAL)
-    interface CompileTimeValue extends Value {
+    interface NonEvaluableValue extends Value {
         @Nullable
         @Override
-        default <M extends Message> Object eval(@Nonnull final FDBRecordStoreBase<M> store,
+        default <M extends Message> Object eval(@Nullable final FDBRecordStoreBase<M> store,
                                                 @Nonnull final EvaluationContext context) {
-            throw new RecordCoreException("value is compile-time only and cannot be evaluated");
+            throw new RecordCoreException("value cannot be evaluated");
         }
     }
 
+    /**
+     * A scalar {@link Value} that can be inverted, i.e. {@code inverse_f(f(x)) = x} for all {@code x}.
+     * @param <V> type parameter of the kind of inverse value of this value.
+     */
+    @API(API.Status.EXPERIMENTAL)
+    interface InvertableValue<V extends Value> extends Value {
+        V createInverseValue(@Nonnull Value newChildValue);
```

Comment:
We cannot really, since what it postulates is that `f` has an inverse `inv` so `inv(f(x)) = x`. What is the result of `inv(...)` if `x` is more than one argument?